app/src/main/java/com/futurice/freesound/feature/search/SearchViewModel.java

```diff
@@ -88,23 +90,23 @@ void openSoundDetails(@NonNull final Sound sound) {
     }
 
     @Override
-    public void bind(@NonNull final CompositeSubscription subscriptions) {
+    public void bind(@NonNull final CompositeDisposable disposables) {
         searchTermRelay.subscribeOn(Schedulers.computation())
                        .observeOn(Schedulers.computation())
                        .map(String::trim)
-                       .switchMap(this::searchOrClear)
+                       .debounce(2, TimeUnit.SECONDS)
+                       .switchMap(it -> searchOrClear(it).toObservable())
```

Comment:
Make `it -> searchOrClear(it).toObservable()` a named method which exposes the `Observable`, rather than the `Maybe`.